lucene/facet/src/java/org/apache/lucene/facet/DrillSideways.java

```diff
@@ -390,18 +398,44 @@ protected boolean scoreSubDocsAtOnce() {
     return false;
   }
 
-  /** Result of a drill sideways search, including the {@link Facets} and {@link TopDocs}. */
+  /**
+   * Result of a drill sideways search, including the {@link Facets} and {@link TopDocs} {@link
+   * FacetsCollector} for drill down and drill sideways passed to {@link #buildFacetsResult} to
+   * compute {@link Facets} are also exposed here as an alternative to subclassing DrillSideways to
+   * obtain them
+   */
   public static class DrillSidewaysResult {
     /** Combined drill down and sideways results. */
     public final Facets facets;
 
     /** Hits. */
     public final TopDocs hits;
 
+    /** FacetsCollector for drill down. */
+    public final FacetsCollector drillDowns;
+
+    /**
+     * Facets Collector for drill sideways Array of FacetsCollector maps directly to Array of
+     * String[] drillSidewaysDims i.e. ith drillSidewaysDims String maps to ith drillSideways
+     * FacetsCollector
+     */
+    public final FacetsCollector[] drillSideways;
+
+    /** drill sideways dimensions */
```

Comment:
This could also be a little more descriptive. Maybe something like:

```suggestion
    /** Dimensions that correspond to to the `drillSideways` FacetsCollectors */
```